src/Sulu/Component/Websocket/ConnectionContext/ConnectionContext.php

```diff
@@ -0,0 +1,196 @@
+<?php
+/*
+ * This file is part of the Sulu CMF.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Websocket\ConnectionContext;
+
+use Guzzle\Http\Message\RequestInterface;
+use Guzzle\Http\QueryString;
+use Ratchet\ConnectionInterface;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBag;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+/**
+ * Simple websocket context
+ */
+class ConnectionContext implements ConnectionContextInterface
+{
+    /**
+     * Return id of connection context for connection
+     * @param ConnectionInterface $conn
+     * @return string
+     */
+    public static function getIdFromConnection(ConnectionInterface $conn)
+    {
+        $dummy = new ConnectionContext($conn);
+
+        return $dummy->getId();
+    }
+
+    /**
+     * @var RequestInterface
+     */
+    private $request = null;
+
+    /**
+     * @var QueryString
+     */
+    private $query = null;
+
+    /**
+     * @var SessionInterface
+     */
+    private $session = null;
+
+    /**
+     * @var ParameterBag
+     */
+    private $parameters;
+
+    /**
+     * @var string
+     */
+    private $id;
+
+    public function __construct(ConnectionInterface $conn)
+    {
+        if (isset($conn->WebSocket)) {
+            $this->request = $conn->WebSocket->request;
+            $this->query = $this->request->getUrl(true)->getQuery();
+        }
+
+        if (isset($conn->Session)) {
+            $this->session = $conn->Session;
+        }
+
+        $sessionName = ini_get('session.name');
+
+        if (isset($this->request) && isset($this->request->getCookies()[$sessionName])) {
+            $this->id = $this->request->getCookies()[$sessionName];
+        } elseif (isset($conn->resourceId)) {
+            $this->id = $conn->resourceId;
+        }
+
+        $this->parameters = new ParameterBag();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getQuery()
+    {
+        return $this->query;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRequest()
+    {
+        return $this->request;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSession()
+    {
+        return $this->session;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getToken($firewall)
+    {
+        if ($this->session !== null) {
+            return unserialize($this->session->get('_security_' . $firewall));
```

Comment:
ne there is no elegant way to do this ... but it is at the moment not used ... should i throw a not implemented exception and add a comment with a possible solution?